### PR TITLE
Add User API Token Generation, Squash some bugs

### PR DESCRIPTION
**Pull request by RossIV**

### CODE REVIEW

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -74,9 +85,9 @@ public function store(Request $request)
                 $user->assignRole($requestedRole);
             }
         }
-
+
```
Comment (kberzinch) [inline app/Http/Controllers/UserController.php]: Trailing whitespace

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -74,9 +85,9 @@ public function store(Request $request)
                 $user->assignRole($requestedRole);
             }
         }
-
+        
         if (is_numeric($user->id)) {
-            $dbUser = User::findOrFail($user->id);
+            $dbUser = User::findOrFail($user->id)->makeVisible('api_token');
```
Comment (kberzinch) [inline app/Http/Controllers/UserController.php]: Why are you doing this here? It's made visible in line 117 after permission checks.

Comment (RossIV) [reply app/Http/Controllers/UserController.php]: Those are two separate functions?

Comment (kberzinch) [reply app/Http/Controllers/UserController.php]: this is a good point

**Pull request merged**